### PR TITLE
Enable connection flow in Jetpack cloud

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This is a proof of concept to enable the connection flow in Jetpack cloud.

### Testing instructions

- Download this PR
- Run Calypso and Jetpack cloud concurrently
- Apply this patch to your sandbox and make `jetpack.wordpress.com` point to it (you'll need to change the port to `3000` if you're not running both platforms concurrently): D53031-code
- Select a self-hosted Jetpack site that's not connected yet
- Visit `/jetpack/connect/` and follow the steps to connect your site
- Verify that through the process you're staying on Jetpack cloud

### CODE REVIEW

**Comment by rcanepa** [comment]
This works nicely. I tested the flow with a site that didn't have the plugin installed and everything worked fine. 

I found just one small detail on the checkout page. If you remove everything from the cart, you will be redirected to `/plans/:site` which is a route that doesn't exist in the cloud.

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@monsieur-z please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/956773). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: client/boot/common.js [hunk 1]
```diff
@@ -127,6 +127,10 @@ const oauthTokenMiddleware = () => {
 			'/connect',
 		];
 
+		if ( isJetpackCloud() && ! isJetpackCloudProd() ) {
```
Comment (monsieur-z) [inline client/boot/common.js]: I'm not a fan of adding an extra environment condition like this, but I think that'll do for a proof of concept.

File: client/my-sites/plans-v2/utils.ts [hunk 2]
```diff
@@ -688,12 +688,12 @@ export function checkout(
...
 		path = `/jetpack/connect/${ productsString }`;
 	} else {
-		path = isJetpackCloud()
+		path = isJetpackCloudProd()
 			? `/checkout/${ siteSlug }/${ productsString }`
 			: `/checkout/${ siteSlug }`;
 	}
 
-	if ( isJetpackCloud() ) {
+	if ( isJetpackCloudProd() ) {
```
Comment (ChaosExAnima) [inline client/my-sites/plans-v2/utils.ts]: Confused: what's going on here? Do we want to disable these for non-prod Cloud?

File: client/boot/common.js (hunk 1 above)
Comment (ChaosExAnima) [reply client/boot/common.js]: If we end up needing this a lot, we should switch to a feature flag imo.

File: client/my-sites/plans-v2/utils.ts (hunk 2 above)
Comment (monsieur-z) [reply client/my-sites/plans-v2/utils.ts]: @ChaosExAnima, in Cloud, except for prod, we'd like the same behaviour as in Calypso, i.e. redirecting to `/checkout/` in the same platform.

**Comment by monsieur-z** [comment]
> This works nicely. I tested the flow with a site that didn't have the plugin installed and everything worked fine.
> 
> I found just one small detail on the checkout page. If you remove everything from the cart, you will be redirected to `/plans/:site` which is a route that doesn't exist in the cloud.

Thanks @rcanepa, I'll add this to the board.

**Comment by monsieur-z** [comment]
I added a feature flag, @ChaosExAnima. Could you give it another review?

**Review by ChaosExAnima — APPROVED** [review]
Tested and works well for me!

**Comment by monsieur-z** [comment]
@Automattic/shilling, could I get a review on this one? Thanks!

**Pull request merged**